### DISCUSSION TITLE
Do we need an SSE heartbeat if `rocket` already sends one?

### DISCUSSION THREAD

**Discussion by luckysori** [get10101/itchysats#2788] [Q&A]
We have a custom heartbeat in the projection feed, but the `rocket` documentation seems to suggest that we don't need one:

```rust
/// ## Heartbeat
///
/// A heartbeat comment is injected into the internal stream and sent at a fixed
/// interval. The comment is discarded by clients and serves only to keep the
/// connection alive; it does not interfere with application data. The interval
/// defaults to 30 seconds but can be adjusted with
/// [`EventStream::heartbeat()`].
```

**Comment by da-kami** [comment]
We have to evaluate what this heartbeat is for.
There is a difference in the API being available and being connected to the SSE feed. 
If the SSE feed becomes unavailable because we e.g. restart the daemon we have to actively subscribe again from the UI. For rocket API calls this does not matter, because after a restart a request against the API will just work, because there is no subscription.

I think the point here is, that we want to react to the `heartbeat` missing on the user interface side.
Can we easily consume the rocket heartbeat in the react code? Is this sufficient to trigger a re-subscription to the feed? 
If so we could fallback to rocket instead of having our own.
